infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -468,8 +481,7 @@ private void fetchTopology() {
      * Stop Log Replication
      */
     private void stopLogReplication() {
-        if (topologyDescriptor.getLocalClusterDescriptor() != null &&
-                topologyDescriptor.getLocalClusterDescriptor().getRole() == ClusterRole.SOURCE && isLeader.get()) {
+        if (isSource() && isLeader.get()) {
```

Comment:
should we have another stopLogReplication API that is per session? cause for the case of full table role switch we only want to stop those specific sessions.

Reply:
Yes, that is there in the connectionModel PR.

I have only taken out the parts which could be easily extracted. While rebasing the connectionModel PR I'll see if I can split it up more.